src/Bitcoin/Script.cpp

```diff
@@ -281,10 +291,15 @@ Script Script::lockScriptForAddress(const std::string& string, enum TWCoinType c
             return buildPayToScriptHash(data);
         }
     } else if (SegwitAddress::isValid(string)) {
-        auto result = SegwitAddress::decode(string);
+        const auto result = SegwitAddress::decode(string);
         // address starts with bc/ltc
-        auto program = std::get<0>(result).witnessProgram;
-        return buildPayToWitnessProgram(program);
+        const auto address = std::get<0>(result);
+        if (address.witnessVersion == 0) {
+            return buildPayToWitnessProgram(address.witnessProgram);
+        }
+        if (address.witnessVersion == 1) {
+            return buildPayToTaprootPublicKey(address.witnessProgram);
```

Comment:
always public key? any other type?

Reply:
Taproot 'hides' everything behind a Schnorr a public key, that can be an aggregated public key

'A Taproot output is a native SegWit output (see BIP141) with version number 1, and a 32-byte witness program. [...]
the witness program [...] represents a public key according to BIP340.'

https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki

Reply:
I don't like the name `buildPayToTaprootPublicKey`, why not add version to `buildPayToWitnessProgram`?

Reply:
That was my first version, I changed it to be more in line with the P2TR script type naming, and emphasize that here a 32-byte public key is needed.
Otherwise it's possible to create v1 with a 20-byte hash, which will not be a spendable.

See https://mempool.space/tx/c24bd72e3eaea797bd5c879480a0db90980297bc7085efda97df2bf7d31413fb, which shows type UNKNOWN (not V0_P2WPKH nor V1_P2TR).

Any better naming?

Reply:
I see, how about `buildPayToV0WitnessProgram` and `buildPayToV1WitnessProgram`?, I guess v1 is only for taproot, if they add new output, will be v2/v3 etc